### PR TITLE
Issue #4981: ImportOrder with order=bottom should disallow non-static import after a static import

### PR DESCRIPTION
**Pull request by julianhyde**
This pull request contains a fix and test case for #4981.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 1]
```diff
@@ -398,6 +398,10 @@ else if (option == ImportOrderOption.BOTTOM) {
             }
             doVisitToken(ident, isStatic, isLastImportAndNonStatic);
 
+            if (isLastImportAndNonStatic) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: If this variable is named correctly, what if the last import is static and it still has a non-static before it?
Example:
```
import a.*;

import static b.f1;
import static c.f2;

import d.*;

import static e.f3;
```
Will your new code still produce a violation on `import d.*` ?

Comment (julianhyde) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: Yes. In 043cd90, I just added a test case that verifies. Feel free to squash it into the previous commit.

**Comment by romani** [comment]
@julianhyde , please make whole fix and one commit 
and please follow guidance from CIs, all CIs should be green.

**Comment by rnveach** [comment]
@julianhyde CI is failing. Please also rebase on latest master.

> [ERROR]   ImportOrderCheckTest.testStaticGroupsAlphabeticalOrderBottomNegative:356->AbstractModuleTestSupport.verify:208->AbstractModuleTestSupport.verify:244->AbstractModuleTestSupport.verify:275 error message 0 expected:<...ottom_Negative.java:[8: Wrong order for 'java.util.Set' import.]> but was:<...ottom_Negative.java:[5: Extra separation in import group before 'java.lang.Math.PI']>
[ERROR]   ImportOrderCheckTest.testStaticGroupsAlphabeticalOrderBottomNegative2:372->AbstractModuleTestSupport.verify:208->AbstractModuleTestSupport.verify:244->AbstractModuleTestSupport.verify:275 error message 0 expected:<...ttom_Negative2.java:[8: Wrong order for 'java.util.Set' import.]> but was:<...ttom_Negative2.java:[5: Extra separation in import group before 'java.lang.Math.PI']>

**Comment by rnveach** [comment]
@julianhyde ping

**Comment by julianhyde** [comment]
Sorry for the delay. I ran into errors in the CI. Some of them might have been mine, others looked like they weren't. Ran out of time to analyze them. Will try to pick this up sometime.

**Comment by romani** [comment]
@julianhyde , 
Please resolve all problems from ImportOrderCheckTest  and rebase on latest master. Problem with translation check should disappear after rebase.

**Comment by romani** [comment]
@julianhyde , ping.

**Comment by romani** [comment]
@julianhyde , do you plan to continue ?
If not, it is completely fine, lets close PR and somebody will pickup your code and finish it.
Open PR is  sign to avoid contributing to issue, so you blocks the issue now.

If no reply in 2 weeks, I will close this PR as abandoned.

**Comment by julianhyde** [comment]
I probably won't get time to work on it. It's frustrating because my code is probably fine, but I don't have time to persuade CI that it is fine. So, if someone else works on this, they are welcome to start with my code.

**Comment by romani** [comment]
Looks like CI problems could be resolved just by rebase on our latest master .  I remember there was such problems a while ago , but they are already resovled.

@djydewang , if still around this Check and have time and interest, please help us to finish this PR.

**Comment by djydewang** [comment]
I am on it.

**Comment by djydewang** [comment]
I am on it.

**Comment by romani** [comment]
closed in favor of #5340

**Pull request closed without merging**